### ISSUE TITLE
clr-installer can start before network is ready

### ISSUE THREAD

**Issue by mdhorn** [clearlinux/clr-installer#252]
**Describe the bug**
Currently, it is possible for the clr-installer to launch before the network is ready. 

**To Reproduce**
- Boot the installer.img file on a system 
- The network may be blank on the Advanced Tab, Configure Network Interface option
  - Start the system with no wired connection active to ensure this scenario

**Expected behavior**
The installer is not useful without a network connection, we would expect the clr-installer service to wait on the network being online.

Adding the following to the Unit section:
```
After=network-online.target
Wants=network-online.target
```
This ensures the network is ready before launching the clr-installer BUT boot-up will hang until there is an active network connection (OR we hit the 90 second timeout before completing the boot). This means we would ASSUME there is an available DHCP service and an active wired connection available to launch the installer image. 

*IS* this reasonable for configuration where the user intends to use a static IP address due to no DHCP server on the network where the installation will take place?

Note: pacdiscovery which is the automatic proxy discovery service also does not currently wait for the network to be ready and hence likey why "sometimes" proxy don't work on boot -- both for the installer as well as generally for Clear Linux OS.

**Comment by chenming1986** [comment]
I think it would be useful for the clr-installer to wait for the network when clri.descriptor is specified in kernel cmdline and points to a network location. Maybe the simplest way for such case is to add several seconds delay before the clr-installer trying to fetch the config file from network?

**Comment by mdhorn** [comment]
With the release of the new live-desktop and live-server ISO images, we no longer automatically launch the installer on boot. 
Also, we have migrated from systemd-networkd to using Network Manager for all wired and wireless connections.

Closing this issue, though we need to review similar functionality when implementing/testing the PXE installation requirements.
https://github.com/clearlinux/clr-installer/issues/247
